src/main/scala/is/hail/asm4s/package.scala

```diff
@@ -26,6 +26,97 @@ package object asm4s {
     def newArray(): AbstractInsnNode
   }
 
+  sealed trait MaybeGenericTypeInfo[T] extends TypeInfo[T] {
+    def castFromGeneric: Code[_] => Code[T]
+    def castToGeneric: Code[T] => Code[_]
+
+    val generic: TypeInfo[_]
```

Comment:
I can't quite get the sense of these variable names.  Generic means primitive but being passed through a Java generic parameter (object)?